### PR TITLE
changed date

### PR DESCRIPTION
**Pull request by kastoestoramadus**

### CODE REVIEW

File: src/main/scala/uk/gov/hmrc/ct/computations/CPQ17.scala [hunk 1]
```diff
@@ -41,5 +41,6 @@ case class CPQ17(value: Option[Boolean]) extends CtBoxIdentifier(name = "Trading
 }
 
 object CPQ17 {
-  val lossReform2017 = LocalDate.parse("2017-04-01")
+  def lossReform2017Applies(to: StartDate): Boolean = to.value.isAfter(lossReform2017)
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/computations/CPQ17.scala]: That's wrong isn't it

It is meant to be EndDate on or after 1/4/2017.

Comment (kastoestoramadus) [reply src/main/scala/uk/gov/hmrc/ct/computations/CPQ17.scala]: :+1: For defining the type. It made the problem to be visible.

File: src/main/scala/uk/gov/hmrc/ct/computations/CPQ17.scala [hunk 2]
```diff
@@ -41,5 +41,6 @@ case class CPQ17(value: Option[Boolean]) extends CtBoxIdentifier(name = "Trading
...
+  private[computations] val lossReform2017 = LocalDate.parse("2017-03-31")
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/computations/CPQ17.scala]: Do we need for the computations package visibility. I think private should be fine.

File: src/main/scala/uk/gov/hmrc/ct/computations/CPQ17.scala (hunk 1 above)
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/computations/CPQ17.scala]: When we change this to EndDate please call the parameter endDate

File: src/main/scala/uk/gov/hmrc/ct/computations/CPQ17.scala [hunk 3]
```diff
@@ -41,5 +41,6 @@ case class CPQ17(value: Option[Boolean]) extends CtBoxIdentifier(name = "Trading
 }
 
 object CPQ17 {
-  val lossReform2017 = LocalDate.parse("2017-04-01")
+  def lossReform2017Applies(to: EndDate): Boolean = to.value.isAfter(lossReform2017)
```
Comment (liquidarmour) [inline src/main/scala/uk/gov/hmrc/ct/computations/CPQ17.scala]: not sure I like to as a parameter name.

Why not call it endDate? Or even apEndDate?

**Pull request merged**